src/dcp/dcp.c

```diff
@@ -271,10 +425,97 @@ int main(int argc, char** argv)
         usage = 1;
     }
 
+    char** argpaths = (&argv[optind]);
+
+#ifdef DAOS_SUPPORT
+    rc = daos_init();
+
+    /* TODO: Don't exit and fail if daos fails init,
+     * could just be regular paths */
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+    }
+
+    struct duns_attr_t dattr  = {0}; 
+    struct duns_attr_t ddattr = {0}; 
+
+    /* Figure out if daos path is the src or dst,
+     * using UNS path, then chop off UNS path
+     * prefix since the path is mapped to the root
+     * of the container in the DAOS DFS mount */
+    if (!daos_uuid_valid(src_pool_uuid) || !daos_uuid_valid(dst_pool_uuid)) {
+        daos_set_paths(&rank, dattr, ddattr, argpaths, &src_pool_uuid, &src_cont_uuid,
+            &dst_pool_uuid, &dst_cont_uuid, dfs_prefix, mfu_src_file, mfu_dst_file);
+    }
+
+    /* check if DAOS source and destination containers are in the same pool */
+    bool same_pool = false;
+    if (mfu_src_file->type == DAOS && mfu_dst_file->type == DAOS) {
+        if (uuid_compare(src_pool_uuid, dst_pool_uuid) == 0) {
+            same_pool = true;
+        }
+    } 
+
+    /* connect to DAOS source pool if uuid is valid */
+    if (mfu_src_file->type == DAOS) {
+       /* if DAOS source pool uuid is valid, then set source file type to DAOS */
+        daos_connect(&rank, &src_poh, &src_coh, &src_pool_uuid, &src_cont_uuid, svc); 
+    }
+
+    if (mfu_dst_file->type == DAOS) {
+        if (daos_uuid_valid(dst_pool_uuid) && !same_pool) {
+            /* if DAOS is the source and destination type, and containers are in different pools,
+             *  then connect to the second pool */
+            daos_connect(&rank, &dst_poh, &dst_coh, &dst_pool_uuid, &dst_cont_uuid, svc); 
+        } else {
+            /* if DAOS is source and destination type, and containers are in the same pool,
+             * then pool is already connected, so we just need to open and/or create the container */
+            if (rank == 0) {
+                /* create container in same pool */
+                daos_cont_info_t co_info;
+                rc = daos_cont_open(src_poh, dst_cont_uuid, DAOS_COO_RW, &dst_coh, &co_info, NULL);
+	        /* If NOEXIST we create it */
+	        if (rc != 0) {
+                    uuid_t cuuid;
+                    rc = dfs_cont_create(src_poh, cuuid, NULL, NULL, NULL);
+                    if (rc != 0) {
+                        MFU_LOG(MFU_LOG_ERR, "Failed to create DFS2 container");
```

Comment:
Should the errors through this section be fatal?

Reply:
@adammoody I think so, since without a container it will not work at all.

Reply:
Ah, I see.  That makes sense.